File: packages/ui-compose/ui/src/test/kotlin/com/kinetixui/ui/blocks/PricingTierBlockTest.kt
package com.kinetixui.ui.blocks

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.width
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.test.junit4.v2.createComposeRule
import androidx.compose.ui.test.onNodeWithText
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.kinetixui.ui.KinetixBadge
import com.kinetixui.ui.KinetixBadgeVariant
import com.kinetixui.ui.KinetixButton
import com.kinetixui.ui.KinetixCard
import com.kinetixui.ui.KinetixCardContent
import com.kinetixui.ui.KinetixCardDescription
import com.kinetixui.ui.KinetixCardFooter
import com.kinetixui.ui.KinetixCardHeader
import com.kinetixui.ui.KinetixCardTitle
import com.kinetixui.ui.KinetixColorScheme
import com.kinetixui.ui.KinetixTheme
import org.junit.Rule
import org.junit.Test
import org.junit.runner.RunWith
import org.robolectric.RobolectricTestRunner
import org.robolectric.annotation.Config

// The "Pricing tier" block — see TestimonialBlockTest.kt for how block fixtures work.
//
// The tick is a text glyph, not an Icon: androidx.compose.material.icons is not on this package's classpath
// (material3 stopped bringing icons-core transitively), and pulling in material-icons-extended to draw one
// check mark would be a real dependency added for an example.
//
// kx-block:start
@Composable
fun PricingTierBlock(onUpgrade: () -> Unit = {}) {
    val colors = KinetixColorScheme.current
    val features = listOf("Unlimited projects", "Priority support", "Custom domains", "Analytics")

    KinetixCard(Modifier.width(300.dp)) {
        KinetixCardHeader {
            KinetixBadge("Most popular", KinetixBadgeVariant.Subtle)
            KinetixCardTitle("Pro")
            KinetixCardDescription("For growing teams.")
            Text("$29", fontSize = 30.sp, fontWeight = FontWeight.SemiBold, color = colors.foreground)
        }
        KinetixCardContent {
            Column(verticalArrangement = Arrangement.spacedBy(8.dp)) {
                features.forEach { feature ->
                    Row(verticalAlignment = Alignment.CenterVertically) {
                        Text("✓", color = colors.primary)
                        Spacer(Modifier.width(8.dp))
                        Text(feature, color = colors.foreground)
                    }
                }
            }
        }
        KinetixCardFooter {
            KinetixButton(onUpgrade, Modifier.fillMaxWidth()) { Text("Upgrade to Pro") }
        }
    }
}
// kx-block:end

@RunWith(RobolectricTestRunner::class)
@Config(sdk = [34])
class PricingTierBlockTest {
    @get:Rule
    val rule = createComposeRule()

    @Test
    fun renders_the_tier_its_features_and_the_call_to_action() {
        rule.setContent { KinetixTheme(darkTheme = false) { PricingTierBlock() } }
        rule.onNodeWithText("Most popular").assertExists()
        rule.onNodeWithText("Pro").assertExists()
        rule.onNodeWithText("Unlimited projects").assertExists()
        rule.onNodeWithText("Analytics").assertExists()
        rule.onNodeWithText("Upgrade to Pro").assertExists()
    }
}

File: packages/ui-flutter/test/blocks/pricing_tier_block_test.dart
import 'package:flutter/material.dart';
import 'package:flutter_test/flutter_test.dart';
import 'package:kinetix_ui/kinetix_ui.dart';

// The "Pricing tier" block — see testimonial_block_test.dart for how block fixtures work.
//
// kx-block:start
class PricingTierBlock extends StatelessWidget {
  const PricingTierBlock({super.key, this.onUpgrade});

  static const List<String> features = <String>[
    'Unlimited projects',
    'Priority support',
    'Custom domains',
    'Analytics',
  ];

  final VoidCallback? onUpgrade;

  @override
  Widget build(BuildContext context) {
    final KinetixColors c = KinetixTheme.of(context);
    return KinetixCard(
      child: Column(
        mainAxisSize: MainAxisSize.min,
        children: <Widget>[
          const KinetixCardHeader(
            children: <Widget>[
              KinetixBadge('Most popular', variant: KinetixBadgeVariant.subtle),
              KinetixCardTitle('Pro'),
              KinetixCardDescription('For growing teams.'),
              Text(r'$29', style: TextStyle(fontSize: 30, fontWeight: FontWeight.w600)),
            ],
          ),
          KinetixCardContent(
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.start,
              children: <Widget>[
                for (final String feature in features)
                  Padding(
                    padding: const EdgeInsets.only(bottom: 8),
                    child: Row(
                      children: <Widget>[
                        Icon(Icons.check, size: 16, color: c.primary),
                        const SizedBox(width: 8),
                        Text(feature),
                      ],
                    ),
                  ),
              ],
            ),
          ),
          KinetixCardFooter(
            children: <Widget>[
              Expanded(
                child: KinetixButton(onPressed: onUpgrade, child: const Text('Upgrade to Pro')),
              ),
            ],
          ),
        ],
      ),
    );
  }
}
// kx-block:end

void main() {
  testWidgets('renders the tier, its features and the call to action', (WidgetTester tester) async {
    await tester.pumpWidget(
      MaterialApp(
        home: KinetixTheme(
          brightness: Brightness.light,
          child: const Scaffold(body: SingleChildScrollView(child: PricingTierBlock())),
        ),
      ),
    );

    expect(find.text('Most popular'), findsOneWidget);
    expect(find.text('Pro'), findsOneWidget);
    expect(find.text('Unlimited projects'), findsOneWidget);
    expect(find.text('Analytics'), findsOneWidget);
    expect(find.text('Upgrade to Pro'), findsOneWidget);
  });
}
